state/applicationoffers_test.go

```diff
@@ -500,16 +517,79 @@ func (s *applicationOffersSuite) addOfferConnection(c *gc.C, offerUUID string) {
 		Username:        "admin",
 		SourceModelUUID: testing.ModelTag.Id(),
 	})
+
+	return app
+}
+
+func (s *applicationOffersSuite) TestRemoveOffersSucceedsWithZeroConnections(c *gc.C) {
+	s.createDefaultOffer(c)
+	ao := state.NewApplicationOffers(s.State)
+	err := ao.Remove("hosted-mysql", false)
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
after removing something you need to refresh and assert that you get a not found error otherwise you're not really testing that the remove worked